enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,209 @@
+        for unenrollment in recent_unenrollments:
+            fulfillment_uuid = unenrollment.get("uuid")
+            enterprise_course_enrollment = unenrollment.get("enterprise_course_enrollment")
+            enrollment_course_run_key = enterprise_course_enrollment.get("course_id")
+            enrollment_unenrolled_at = enterprise_course_enrollment.get("unenrolled_at")
+            try:
+                # Look for a transaction related to the unenrollment
+                # This will throw a Transaction.DoesNotExist exception if no transaction is found
+                related_transaction = Transaction.objects.get(uuid=unenrollment.get('transaction_id'))
+                idempotency_key = generate_transaction_reversal_idempotency_key(
+                    fulfillment_uuid,
+                    enrollment_unenrolled_at
+                )
+                # This ORM query will throw a Reversal.DoesNotExist exception if no reversal is found,
+                # which means we can move forward with the reversal generation
+                existing_reversal = Reversal.objects.get(
+                    transaction=related_transaction,
+                    idempotency_key=idempotency_key,
+                )
+            except Transaction.DoesNotExist:
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}No Subsidy Transaction found for enterprise fulfillment: {fulfillment_uuid}"
+                )
+                continue
+            except Reversal.DoesNotExist:
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}No existing Reversal found for enterprise fulfillment: {fulfillment_uuid}. "
+                    f"Writing Reversal for Transaction: {related_transaction}."
+                )
+
+                # On initial release we are only supporting learner initiated unenrollments for OCM courses.
+                # OCM courses are identified by the lack of an external_reference on the Transaction object.
+                # Externally referenced transactions can be unenrolled through the Django admin actions related to the
+                # Transaction model.
+                if not related_transaction.external_reference.exists():
+                    # Memoize the content metadata for the course run fetched from the enterprise catalog
+                    if not fetched_content_metadata.get(enrollment_course_run_key):
+                        try:
+                            customer_uuid = related_transaction.ledger.subsidy.enterprise_customer_uuid
+                        except Subsidy.DoesNotExist:
+                            logger.warning(
+                                f"{DRY_RUN_LOG_PREFIX}Unable to determine enterprise customer uuid for transaction: "
+                                f"{related_transaction}"
+                            )
+                            continue
+
+                        content_metadata = enterprise_catalog_api_client.get_content_metadata_for_customer(
```

Comment:
use the `ContentMetadataApi.get_content_metdata()` function here, which will use the cache for you.